src/Workspaces/Core/Portable/Execution/AbstractOptionsSerializationService.cs

```diff
@@ -204,5 +210,112 @@ protected void WriteParseOptionsTo(ParseOptions options, ObjectWriter writer, Ca
 
             features = featuresList ?? SpecializedCollections.EmptyEnumerable<KeyValuePair<string, string>>();
         }
+
+        protected void WriteOptionSetTo(OptionSet options, string language, ObjectWriter writer, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // order of options we serialize should be static so that we get deterministic checksum for options
+            WriteOptionTo(options, language, CodeStyleOptions.QualifyFieldAccess, writer, cancellationToken);
```

Comment:
i'm confused.  why are there two sets of option writing and option reading code?